src/install/installer.ts

```diff
@@ -491,10 +490,12 @@ export class Installer {
     return pkgUrls;
   }
 
-  private getBestExistingMatch(matchPkg: PackageTarget): ExactPackage | null {
+  private async getBestExistingMatch(
+    matchPkg: PackageTarget
+  ): Promise<ExactPackage | null> {
     let bestMatch: ExactPackage | null = null;
     for (const pkgUrl of this.pkgUrls) {
-      const pkg = this.resolver.parseUrlPkg(pkgUrl);
+      const pkg = await this.resolver.parseUrlPkg(pkgUrl);
```

Comment:
If we use the version field for the hash, we should be able to syncify this again.

Reply:
Yeah, makes sense, I'll add some documentation about the intention for `ExactPackage` - that it should pin down a package uniquely against a registry. From that point of view `version` is the right field for the hash.